iceberg/src/main/scala/org/apache/spark/sql/delta/icebergShaded/HudiConverter.scala

```diff
@@ -0,0 +1,363 @@
+          snapshotToConvert.deltaLog,
+          "delta.hudi.conversion.skipped.emptyCatalogTable",
+          data = Map(
+            "version" -> snapshotToConvert.version
+          )
+        )
+      None
+    }
+  }
+
+  /**
+   * Convert the specified snapshot into Hudi. NOTE: This operation is blocking. Call
+   * enqueueSnapshotForConversion to run the operation asynchronously.
+   * @param snapshotToConvert the snapshot that needs to be converted to Hudi
+   * @param txnOpt the OptimisticTransaction that created snapshotToConvert.
+   *            Used as a hint to avoid recomputing old metadata.
+   * @param catalogTable the catalogTable this conversion targets
+   * @return Converted Delta version and commit timestamp
+   */
+  private def convertSnapshot(
+      snapshotToConvert: Snapshot,
+      txnOpt: Option[OptimisticTransactionImpl],
+      catalogTable: CatalogTable): Option[(Long, Long)] =
+      recordFrameProfile("Delta", "HudiConverter.convertSnapshot") {
+    val log = snapshotToConvert.deltaLog
+    val metaClient = loadTableMetaClient(snapshotToConvert.deltaLog.dataPath.toString,
+      catalogTable.identifier.table, snapshotToConvert.metadata.partitionColumns,
+      log.newDeltaHadoopConf())
+    val lastDeltaVersionConverted: Option[Long] = loadLastDeltaVersionConverted(metaClient)
+    val maxCommitsToConvert =
+      spark.sessionState.conf.getConf(DeltaSQLConf.ICEBERG_MAX_COMMITS_TO_CONVERT)
+
+    // Nth to convert
+    if (lastDeltaVersionConverted.exists(_ == snapshotToConvert.version)) {
+      return None
+    }
+
+    // Get the most recently converted delta snapshot, if applicable
+    val prevConvertedSnapshotOpt = (lastDeltaVersionConverted, txnOpt) match {
+      case (Some(version), Some(txn)) if version == txn.snapshot.version =>
+        Some(txn.snapshot)
+      // Check how long it has been since we last converted to Hudi. If outside the threshold,
+      // fall back to state reconstruction to get the actions, to protect driver from OOMing.
+      case (Some(version), _) if snapshotToConvert.version - version <= maxCommitsToConvert =>
+        try {
+          // TODO: We can optimize this by providing a checkpointHint to getSnapshotAt. Check if
+          //  txn.snapshot.version < version. If true, use txn.snapshot's checkpoint as a hint.
+          Some(log.getSnapshotAt(version))
+        } catch {
+          // If we can't load the file since the last time Hudi was converted, it's likely that
+          // the commit file expired. Treat this like a new Hudi table conversion.
+          case _: DeltaFileNotFoundException => None
+        }
+      case (_, _) => None
+    }
+
+    val tableOp = (lastDeltaVersionConverted, prevConvertedSnapshotOpt) match {
+      case (Some(_), Some(_)) => WRITE_TABLE
+      case (Some(_), None) => REPLACE_TABLE
+      case (None, None) => CREATE_TABLE
+    }
+
+    UniversalFormat.enforceSupportInCatalog(catalogTable, snapshotToConvert.metadata) match {
```

Comment:
plz remove this part for now... this is to specifically update HMS with Iceberg props.

Reply:
Removed